lib/templates/accountTemplate.js

```diff
@@ -86,23 +95,59 @@ class AccountTemplate {
     fsUtils.writeConfig(this.TEMPLATE_TYPE, name, templateConfig);
   }
 
-  static #prepareConfigDetails(template, existingConfig = {}) {
+  static #prepareConfigDetails(type, envId, template, existingConfig = {}) {
     const attributes = this.CONFIG_ITEMS.reduce((acc, attribute) => {
       if (template.hasOwnProperty(attribute)) {
         acc[attribute] = template[attribute];
       } else if (existingConfig?.hasOwnProperty(attribute)) {
         acc[attribute] = existingConfig[attribute];
       }
+
       return acc;
     }, {});
+
+    // We need to also create the mapping_list_ranges, which could be different per firm & partner
+    const existingMappingRanges = existingConfig.mapping_list_ranges || [];
+    const updatedMappingRanges = [...existingMappingRanges];
+
+    for (let mappingListRange of template.mapping_list_ranges) {
+      // Find the existing mapping list range, if any
+      const existingIndex = existingMappingRanges.findIndex(
+        (existingMappingListRange) =>
+          existingMappingListRange.account_mapping_list_id ===
+          mappingListRange.account_mapping_list_id
+      );
+
+      if (existingIndex !== -1) {
+        // If it exists, retain existing properties and update "type" and "env_id"
+        updatedMappingRanges[existingIndex] = {
```

Comment:
If it exists, it might still have had the actual range updated, so the local version should be updated too, no?

Reply:
Good catch; updated